### PR TITLE
Use nixpkgs-unstable

### PR DESCRIPTION
**Pull request by Hoverbear**
##### Description

So it sounds like there is no unified channel for Linux and Mac for nixpkgs in any stable form, only unstable.

If you're on Mac you have to pick `nixpkgs-22.05-darwin`  (which is undocumented https://nixos.wiki/wiki/Nix_channels) and on Linux you have to pick `nixos-22.05` since `nixpkgs-22.05` does not exist.

Opting to use `nixpkgs-unstable` means we may see some breakage, but we can always opt not to update our lock and keep our dependencies unchanged.

This creates a better experience for Mac users trying to build `nix-installer` as they don't need to build the whole world.

##### Checklist

- [ ] Formatted with `cargo fmt`
- [ ] Built with `nix build`
- [ ] Ran flake checks with `nix flake check`
- [ ] Added or updated relevant tests (leave unchecked if not applicable)
- [ ] Added or updated relevant documentation (leave unchecked if not applicable)
- [ ] Linked to related issues (leave unchecked if not applicable)

##### Validating with `install.determinate.systems`

If a maintainer has added the `upload to s3` label to this PR, it will become available for installation via `install.determinate.systems`:

```shell
curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix/pr/$PR_NUMBER | sh -s -- install
```

### CODE REVIEW

**Review by cole-h — APPROVED** [review]
Well, it would be 22.11 now ;) And you can see a list of all currently maintained channels on https://status.nixos.org/.

Any potential breakage will likely be limited to builds and not exposed to users. As far as I know, the only real difference between nixpkgs-unstable and nixos-unstable is that nixos- adds more nixos-specific tests (e.g. so a kernel/btrfs upgrade doesn't render btrfs users' systems unbootable, or that a podman upgrade didn't break things).

So basically I think there's nothing for downstream users to worry about, and anything that does break should be caught by CI and/or our tests on PRs that bump Nixpkgs.
